### PR TITLE
refactor(contract): move shared request and auth plumbing into api/common.rs

### PR DESCRIPTION
**Pull request by gilcu3**
Part 2/12 of #1962

### CODE REVIEW

**Review by pbeza — DISMISSED** [review]
I won't lie, Claude helped me confirm this is a mechanical refactor. Only these changes aren't mechanical (caught by Claude), but they look good:

> - Visibility widened from private to `pub(crate)` on the moved items, which the move requires and which changes nothing externally
> - Doc-comment-only edits: intra-doc links rewritten to full paths (`crate::state::ProtocolContractState::...`) so they resolve from the new module, and one added clause on `MINIMUM_SIGN_REQUEST_DEPOSIT` ("also charged for foreign-transaction verification")
> - Import shuffling in `lib.rs` (drops now-unused `GasWeight` and `RequestError`, imports the moved items from `api::common`) and the new file's own imports, module doc line, impl-block/test-module scaffolding, and closing braces

File: crates/contract/src/api/common.rs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        env::panic_str(&RequestError::Timeout.to_string());
+    }
+}
+
+/// Register used to receive data id from `promise_await_data`.
+/// Note: This is an implementation constant, not a configurable policy value.
+const DATA_ID_REGISTER: u64 = 0;
+
+/// Minimum deposit required for sign requests, also charged for foreign-transaction verification
+pub(crate) const MINIMUM_SIGN_REQUEST_DEPOSIT: NearToken = NearToken::from_yoctonear(1);
```
Comment (kevindeforth) [inline crates/contract/src/api/common.rs]: Lets move this to the contract-interface crate.

File: crates/contract/src/api/common.rs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    #[private]
+    pub fn fail_on_timeout() {
+        // To stay consistent with the old version of the timeout error
+        env::panic_str(&RequestError::Timeout.to_string());
+    }
+}
+
+/// Register used to receive data id from `promise_await_data`.
+/// Note: This is an implementation constant, not a configurable policy value.
+const DATA_ID_REGISTER: u64 = 0;
```
Comment (kevindeforth) [inline crates/contract/src/api/common.rs]: I generally prefer when we define constant at the top of the file.
Also, I would suggest to move this to a separate file together with `enqueue_yield_request`.

File: crates/contract/src/api/common.rs [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+        assert_eq!(
+            signer_id, predecessor_id,
+            "Caller must be the signer account (signer: {}, predecessor: {})",
+            signer_id, predecessor_id
+        );
+
+        signer_id
+    }
+}
```
Comment (kevindeforth) [inline crates/contract/src/api/common.rs]: Have you considered moving these into a separate file concerned solely with authorization? Something like`auth.rs` or `authorization_helpers.rs` or `access_control.rs`?

File: crates/contract/src/api/common.rs [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+/// Transfers `amount` to `account_id` via a detached promise; no-op when zero.
+pub(crate) fn refund_to(account_id: &AccountId, amount: NearToken) {
+    if amount > NearToken::from_near(0) {
+        log!("refund {amount} to {account_id}");
+        Promise::new(account_id.clone()).transfer(amount).detach();
+    }
+}
```
Comment (kevindeforth) [inline crates/contract/src/api/common.rs:153]: Hmm, I think it wouldn't be bad to have maybe one file where we track all the promises that this contract spawns (e.g. `promises.rs`).

This method and `enqueue_yield_request` kind of belong together in that sense.

**Review by kevindeforth — COMMENTED** [review]
I would suggest to split this into even more files.

**Review by kevindeforth — DISMISSED** [review]
I pre-approved, because it's an improvement and I don't want to block, but I would still suggest to consider splitting this further.

File: crates/contract/src/api/common.rs (hunk 4 above)
Comment (gilcu3) [reply crates/contract/src/api/common.rs:153]: Unfortunately we have many more methods using promises than this two, so it would not be very uniform to have them being located there. Leaving here for now, we could think later if we want to do some promise unification

**Review by kevindeforth — APPROVED** [review]
Thanks!

**Review by pbeza — APPROVED** [review]
